### PR TITLE
Add WGSL intrinsics for synchronization

### PR DESCRIPTION
**Pull request by aleino-nv**
This addresses issue #5085.

### CODE REVIEW

**Comment by aleino-nv** [comment]
@csyonghe I'm a bit unsure about how the WGSL memory model maps to Slang/HLSL.

Does Slang adopt HLSL's memory model?

**Comment by csyonghe** [comment]
Slang currently has no official memory model. The behavior around atomics and synchronization is mostly target dependent right now. HLSL is also kind of loose in the front in that there isn't a clear spec on what the memory model is there.

**Review by csyonghe — APPROVED** [review]

**Comment by csyonghe** [comment]
Please use "Closes #5085" in the description, so that issue can automatically be closed by github when this PR lands.

**Comment by aleino-nv** [comment]
> Please use "Closes #5085" in the description, so that issue can automatically be closed by github when this PR lands.

Ok I changed "addresses" to "closes".

File: source/slang/hlsl.meta.slang [hunk 1]
```diff
@@ -7553,11 +7555,12 @@ void DeviceMemoryBarrier()
         {
             OpMemoryBarrier Device AcquireRelease|UniformMemory|ImageMemory;
         };
+    case wgsl: __intrinsic_asm "storageBarrier(); textureBarrier(); workgroupBarrier();";
```
Comment (jkwak-work) [inline source/slang/hlsl.meta.slang:7558]: Do we need workgroupBarrier here?

**Review by jkwak-work — COMMENTED** [review]
I think we need test for those implementation.

**Pull request merged**